src/index.js

```diff
@@ -1,50 +1,114 @@
 import './style.css';
 
-const ToDoTask = document.getElementById('ToDoTask');
-
-const toDoslist = [
-  {
-    description: 'wash car',
-    completed: 'false',
-    index: '1',
-  },
-  {
-    description: 'complet project',
-    completed: 'false',
-    index: '2',
-  },
-  {
-    description: 'go to Gym',
-    completed: 'false',
-    index: '3',
-  },
-];
-
-ToDoTask.innerHTML = `
-  <article>
-    <h2>Today's todo </h2>
-    <iconify-icon icon="bx:refresh"></iconify-icon>
-  </article>
-  <hr>
-  <form>
-    <input type="text" id="todo" class="todo-input" placeholder="Add to your list...">
-    <button type="submit" id="addBtn" value><iconify-icon icon="uil:enter"></iconify-icon></button>
-  </form>
-  <hr>
-`;
-
-function showtodos() {
-  toDoslist.forEach((toDo) => {
-    const listItem = document.createElement('li');
-    listItem.classList.add('task');
-    listItem.innerHTML = ` 
-      <input type="checkbox">
-      <label>${toDo.description}</label>
-      <iconify-icon icon="ph:dots-three-outline-vertical-fill" class="dots"></iconify-icon>
-      <hr>  
-    `;
-
-    ToDoTask.append(listItem);
-  });
+const todoInput = document.querySelector('.todo-input input');
+const taskBox = document.querySelector('.task-box');
+const clearAll = document.querySelector('.clear');
+const filters = document.querySelectorAll('.filters span');
+let editId;
+let isEditTask = false;
+let todos = JSON.parse(localStorage.getItem('todo-list'));
```

Comment:
- [ ] Dear @Yemisirach, as the previous reviewer mentioned, the property index is required to be set to the value of the new array length. And if an item is deleted that should update all remaining items' indexes. currently all your task are having index of `0.`

Reply:
I do this part. you can check on local storage.